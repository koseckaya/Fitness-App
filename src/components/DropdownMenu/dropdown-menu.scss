.dropdown {
  position: relative;
}

.dropdown-menu {
  position: absolute;
  right: 0;
  width: 200px;
  list-style-type: none;
  margin: 5px 0;
  padding: 20px 10px;
  color: rgba(0,0,0,.85);
  background-color: #fff;
  line-height: 22px;
  box-shadow: rgb(115 136 189 / 15%) 0px 4px 12px;
  border-radius: 6px;
  z-index: 1;
}

.dropdown-menu > li {
  margin: 0;
  background-color: white;
  line-height: 22px;
  cursor: pointer;
  transition: all .3s;
}

.dropdown-menu > li:hover {
  background-color: #f5f5f5;
}

.dropdown-menu > li > a,
.dropdown-menu > li > button {
  display: block;
  width: 100%;
  height: 100%;
  text-align: left;
  background: none;
  color: inherit;
  border: none;
  padding: 5px;
  margin: 0;
  font: inherit;
  cursor: pointer;
}

.dropdown {
  & .button {
    width: auto;
  }  
}
