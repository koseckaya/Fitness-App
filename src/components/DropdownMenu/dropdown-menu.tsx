import { User } from "firebase/auth";
import { FC, useState } from "react";
import { Link, useNavigate } from "react-router-dom";
import { Button } from "../Button";
import { signOutUser } from "../utils/firebase/firebase";
import useOuterClick from "../utils/useOuterClick";

import './dropdown-menu.scss'

type MenuProps = {
  open: boolean,
  menu: React.ReactNode[],
  trigger: React.ReactNode
};

type DropdownProps = {
  userName: User
}

export const DropdownMenu: FC<DropdownProps> = ({userName}: DropdownProps) => {
  const [open, setOpen] = useState(false);
  const navigate = useNavigate();
  
  const handleOpen = () => {
    setOpen(!open);
  }

  const signOut = () => {
    console.log('sign out')
    setOpen(false);
    signOutUser();
    navigate('/')
  }

  const ref = useOuterClick<HTMLDivElement>(
    (e: Function ) => { setOpen(false) });

  return (
    <div ref={ref}>
      <DropdownItem
        open={open}
        trigger={<Button content={`${userName.email}`} 
        type={Button.TYPES.DEFAULT} 
        onClick={handleOpen}/>}
        menu={[
          <Link to={`/profile`}>
            Profile
          </Link>,
          <button onClick={signOut}>Sign out</button>,
        ]}
      />
    </div>
  );
}

const DropdownItem = ({ open, trigger, menu }: MenuProps) => {
  return (
    <div className="dropdown">
      {trigger}
      {open ? (
        <ul className="dropdown-menu">
          {menu.map((menuItem, index) => (
            <li key={index}>{menuItem}</li>
          ))}
        </ul>
      ) : null}
    </div>
  );
};