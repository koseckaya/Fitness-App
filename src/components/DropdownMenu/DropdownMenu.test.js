import React from 'react';
import renderer from 'react-test-renderer';
import { DropdownMenu } from './dropdown-menu';

const props = {
    displayName: "Mila",
    email:"interika11@gmail.com",
    emailVerified:true,
}


describe("DropDownMenu component", () => {
    it('should render component with default props', () => {
        const component = renderer
            .create(<DropdownMenu {...props}></DropdownMenu>)
            .toJSON();
        expect(component).toMatchSnapshot();
    });
    it('should render component without props', () => {
        const newProps = { ...props, displayName: '', email: '', emailVerified: false }
        const component = renderer
            .create(<DropdownMenu {...newProps}></DropdownMenu>)
            .toJSON();
        expect(component).toMatchSnapshot();
    });
})