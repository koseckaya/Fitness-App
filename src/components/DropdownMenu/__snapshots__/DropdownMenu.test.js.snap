// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropDownMenu component should render component with default props 1`] = `
<div
  className="profileBtnContainer"
>
  <div
    className="dropdown"
  >
    <div
      className="button  "
      onClick={[Function]}
    />
  </div>
</div>
`;

exports[`DropDownMenu component should render component without props 1`] = `
<div
  className="profileBtnContainer"
>
  <div
    className="dropdown"
  >
    <div
      className="button  "
      onClick={[Function]}
    />
  </div>
</div>
`;
